### ISSUE TITLE
Reduce the number of the agent's RPC calls 

### ISSUE THREAD

**Issue by fredo** [beamer-bridge/beamer#778]
We need to cut down the number of RPC calls as costs are very high for professional endpoints.

**Comment by istankovic** [comment]
@fredo do we have an idea of what we would like to do here to further reduce the calls?
If not, then I suggest we either have a brainstorming on the topic and create new stories here, or close this epic.

**Comment by fredo** [comment]
this is a perfect topic for this week's workshop. there are a bunch of RPC calls which are called by the agent during processing state or events. might be possible to remove them.
let's take a bit of time and look over the code. 
before that we should analyze the live angents' behaviour and determine prioritization. (if the live agents used a reasonable amount of calls , we might postpone this)

**Comment by andfletcher** [comment]
@fredo @istankovic have you created issues in your workshop week that we can add to this epic?

**Comment by fredo** [comment]
We did not come to the point talking about this

**Comment by andfletcher** [comment]
once #652 is done we should recheck if we need to actually do this.
